core/src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -4936,6 +4932,109 @@ abstract class MergeIntoSuiteBase
     expectErrorContains = "Cannot add column 'value.a.z' with type 'void'",
     expectErrorWithoutEvolutionContains = "All nested columns must match")
 
+  testNestedStructsEvolution("new source column in map struct")(
+    target = """{ "key": "A", "map": { "key": { "a": 1 } } }""",
```

Comment:
You can add in each test a row `C` in the target and not present in the source to cover behaviour for rows that are not updated